orchestration/dagster_orchestration/hca_manage/validation.py

```diff
@@ -0,0 +1,65 @@
+from google.cloud import storage
+import logging
+from jsonschema import validate
+import json
+import requests
+from typing import Optional
+from hca_manage.common import DefaultHelpParser
+
+
+def validate_json(blob: storage.Blob) -> Optional[Exception]:
+    """
+    Validate that the JSON file blob follows the schema in the describedBy
+    :param blob: JSON file blob to check
+    :return: An exception if JSON is invalid, return None if valid
+    """
+    try:
+        data = json.loads(blob.download_as_string(client=None))
+        schema_path = data["describedBy"]
+        raw_schema = requests.get(schema_path).json()
```

Comment:
🔧 This is going to be expensive if we have to pull down the schema for every json file we validate (there's not that many). 

What if we create a top level method dict that can cache schema bodies and we can refer to that?


```suggestion

class SchemaFetcher:
    def __init__(self):
        self._schema_cache = {}

    def fetch_schema(self, path: str):
        if path in self._schema_cache:
            return self._schema_cache[path]

        raw_schema = requests.get(schema_path).json()
        self._schema_cache[path] = raw_schema
        return raw_schema



...


def validate_json(blob: storage.Blob, schema_fetcher: SchemaFetcher) -> Optional[Exception]:

...

schema = schema_fetcher.fetch_schema(schema_path)
          

```